### ISSUE TITLE
Carousel: Frequently changed active slide and press navigation button crash the control

### ISSUE THREAD

**Issue by mazavr** [angular-ui/bootstrap#5178]
Chrome 47.0.2526.106 m
Please, see http://plnkr.co/edit/mk5Pby?p=preview

User frequently clicks on links "go to slide ##", sometimes user see two or three slides at the same time. 
When user see few slides he clicks on navigation button that appears above the links (left arrow, that means go to previous slide).
If user repeat these action few times the control will be broken.

**Comment by wesleycho** [comment]
Is this present in an older version?

**Comment by mazavr** [comment]
I've used 1.0.0 version in my example. Previous version breaks when user clicks the links very often (without clicking navigation button at all). Now if user clicks only links... few slides appears as well, but one of them disappears after some time.

**Comment by wesleycho** [comment]
Ah I see - yep, this is an issue. I created an imperfect fix in 1.0 that fixed it from completely breaking, but there is a problem here still.

Thanks for the clear reproduction.

**Comment by mazavr** [comment]
Full break of the control might be solved by adding `nextSlide.active = false` next to place a slide to a buffer

```    
    //Prevent this user-triggered transition from occurring if there is already one in progress
    if (nextSlide && nextSlide !== self.currentSlide && !$scope.$currentTransition) {
      goNext(nextSlide, nextIndex, direction);
    } else if (nextSlide && nextSlide !== self.currentSlide && $scope.$currentTransition) {
      bufferedTransitions.push(nextSlide);
      nextSlide.active = false;  // <--- Here
    }
```
